pkg/config/analysis/local/istiod_analyze.go

```diff
@@ -281,7 +281,7 @@ func (sa *IstiodAnalyzer) AddRunningKubeSourceWithRevision(c kubelib.Client, rev
 		// better to fail fast, and get a good idea for the failure.
 		scope.Analysis.Errorf("Failed to watch crd resource for analysis: %s", err)
 	})
-	if err != nil {
+	if err != nil && !ignoreSetWatchError {
```

Comment:
This seems like either a code smell OR something that is under-documented. Ignoring error is generally a bad idea, so if we should be doing it we need some explanation in the comments.

However, it does seem odd to do this. It seems like it would be better to have some smarter logic about setting it only once rather than ignoring errors. check before we attempt rather than ignoring error

Reply:
Why could we met SetWatchErrorHandler return error?

Reply:
In Istio-Operator pod, `Reconcile` is called more than once with a share client.

Reply:
IC when the informer has been started, the SetWatchErrorHandler will return error. 

So can we check and SetWatchErrorHandler when the informer has not been started. I think this way is better than passing a bool

Reply:
there's no proper function on `cache.SharedInformer`, `HasStarted` can not be called,  `HasSynced` seems not suitable for this scenario.

Reply:
after talking with @hzxuzhonghu offline, will find a better way to resolve this.